content/klc/G2.1.adoc

```diff
@@ -1,21 +1,29 @@
 +++
-brief = "Defining generic and atomic parts"
+brief = "Defining generic and fully specified symbols"
 +++
 
-In general terms, the _symbols_ in the KiCad library can be categorized into two groups:
+KiCad generally has two workflows with regards to assign a footprint to a symbol. It can be done in the library or using the "assign footprints tool" at the end of the design process.
 
 **Generic symbols**
 
-Generic symbols can be used with multiple footprints, and _do not_ have a default footprint assigned. Generic symbols allow flexibility in the design workflow. Symbols are first added to the schematic, and the footprint is selected before PCB layout. Using generic symbols allows a small number of library symbol elements to represent a _very large number_ of possible component combinations.
+Generic symbols can be used with multiple footprints, and _do not_ have a default footprint assigned. Generic symbols allow flexibility in the design workflow. For this workflow the symbol placed into the schematic is typically selected to fit the intended function. The footprint assignment happens later when the exact part number is selected. (At the latest when starting with the layout) Using generic symbols allows a small number of library symbol elements to represent a _very large number_ of possible component combinations.
 
-An example of the use of generic symbols is the **Resistor** and **Capacitor** symbols available in the KiCad library. These symbols do not have an assigned default footprint, as there are _many_ possible compatible footprints.
+An example of the use of generic symbols are the **Resistor** and **Capacitor** symbols available in the KiCad library. These symbols do not have an assigned default footprint, as there are _many_ possible compatible footprints.
 
 {{< klcimg src="G2.1a" title="Single schematic symbol" >}} {{< klcimg src="G2.1b" title="Associated with multiple footprints" >}}
 
-**Atomic Parts**
+**Fully specified symbols**
 
-Atomic parts fully define a component, specifying a matching footprint, and are named based on the `MPN` (manufacturer part number). Atomic parts are ready to be placed onto the PCB as they are already associated with a footprint.
+Fully specified symbols define a matching footprint, and are named based on the `MPN` (manufacturer part number). Such symbols do not depend on the user to select the correct footprint later but can have alternative footprints defined using footprint filters such that the user can select an alternate footprint.
 
 {{< klcimg src="G2.1c" title="Atomic symbol" >}}
 
+Some users refer to fully specified symbols as atomic parts or a library of such components as atomic library. Atomic implies a one to one inseparable connection between assets. The closest that KiCad can get to is a fully specified symbol that points to a footprint specialized to the same component.
```

Comment:
This implies to me there is a limitation of KiCad, when it's really the system defined by KLC. A fully atomic library would be possible in KiCad. Can this be re-worded to solely focus on the library resulting from the rules defined in KLC?

Reply:
The inseparatable part is a limitation of kicad as the symbol and footprint libs are indeed separate. (Even if a symbol points to a footprint there is nothing that ensures that the same footprint is not used by another symbol. A true atomic entity would ensure this. At least if you go by the strictest definition of the word as used for example when talking about atomic operations of a CPU.)

Reply:
Umm...

Anyway, maybe I should say I like if the note about KiCad is removed and instead the doc just focuses on the library system defined by KLC. You are contrasting the perception of the word 'atomic' to the terms and library given by following KLC. I guess it seems subjective and interpretive to me instead of just getting down to business and stating what KLC is intended to accomplish in this section.

Please add 'an' before 'atomic library' in the first sentence.

Reply:
The reason why this section even exists is because the terms used are sadly not standardized. Every user has a slightly different interpretation of them depending on which field or software they come from. So i really need to define the terms and how exactly they are used in the context of the KLC as it will otherwise lead to misinterpretation of the KLC.
As far as i understand it altium for example has true atomic libs which is why i also highlight why it is not possible with kicads data structure. I can remove this one part (the stuff about why it is not possible to make true atomic libs) but i am not prepared to remove the definition of terms.

Reply:
I massively reworded this section and made it cursive to make it clear that this is a sidenote (that mainly deals with the confusing use of atomic symbols and atomic parts to refer to very different things.)

Reply:
Alright. Please add a comma after `Confusingly` and we'll call it good.

Reply:
done